pkg/controller/jobs/job/job_controller.go

```diff
@@ -63,20 +62,22 @@ func init() {
 	}))
```

Comment:
What if `IntegrationCallbacks` stores `NewJob` and `NewWorkloadHandler`. Then we don't need a "factory".

Also, do we need both `NewJob` and `JobType`?

Reply:
It will be harder to follow in my opinion. And even if I don't know if we'll eve need it in the future ...  we can register 
another custom reconciler if needed.